src/nodes.coffee

```diff
@@ -1279,12 +1279,18 @@ exports.Class = class Class extends Base
       result
 
   compileClassDeclaration: (o) ->
-    @ctor ?= @makeDefaultConstructor() if @externalCtor
+    @ctor ?= @makeDefaultConstructor() if @externalCtor or @boundMethods.length
     @ctor?.noReturn = true
 
+    @proxyBoundMethods o if @boundMethods.length
+
     o.indent += TAB
 
     result = []
+    if @assignParentRef
```

Comment:
Whilst this is a nice way of doing (so I'm not suggesting changing it), but the [`.cache`](https://github.com/jashkenas/coffeescript/blob/master/src/nodes.coffee#L105) mechanism could be used to instead generate code like:

```
class Child extends (ref = parent.class().expression) {}
```